src/callgraphwidget.cpp

```diff
@@ -62,6 +70,23 @@ CallgraphWidget::~CallgraphWidget() = default;
 
 CallgraphWidget *CallgraphWidget::createCallgraphWidget(const Data::CallerCalleeResults results, QWidget *parent)
 {
+#if KCOREADDONS_VERSION >= QT_VERSION_CHECK(5, 86, 0)
+    const KPluginMetaData md(QStringLiteral("kgraphviewerpart"));
+
+    const auto result = KPluginFactory::instantiatePlugin<KParts::ReadOnlyPart>(md, parent, {});
+
+    if (!result) {
+        return nullptr;
+    }
+
+    auto interface = qobject_cast<KGraphViewer::KGraphViewerInterface*>(result.plugin);
```

Comment:
from here on downwards, duplicates code, introduce a `QObject *loadPlugin(const QString &name);` which does the needed stuff internally